### ISSUE TITLE
m2m upsert in pivot table

### ISSUE THREAD

**Issue by Thomas-git** [Vincit/objection.js#1820]
Hi,
I'm running into a problem when I upsert data into a pivot table when the pivot table use a composite key.

Users relation mapping
`static get relationMappings () {
    return {
      providers: {
        relation: Model.ManyToManyRelation,
        modelClass: IdentityProviders,
        join: {
          from: 'users.id',
          through: {
            from: 'users_identity_providers.user_id',
            to: 'users_identity_providers.ip_id',
            modelClass: UsersIdentityProviders,
            extra: ['provider_id', 'url']
          },
          to: 'identity_providers.id'
        }
    }
  }`
users_identity_providers.ip_id and users_identity_providers.user_id is the composite primary key

When I do a user upsert with data:
`{
  id:1,
          providers: [{
            provider_id: 'TESTID',
            ip_id: 6
          }]
        }`

It fails with:

DBError: select "users_identity_providers"."user_id" as "objectiontmpjoin0", "users_identity_providers"."provider_id" as "provider_id", "identity_providers"."ip_id", "identity_providers"."id" from "identity_providers" inner join "users_identity_providers" on "identity_providers"."id" = "users_identity_providers"."ip_id" where "users_identity_providers"."user_id" in ($1) - column identity_providers.ip_id does not exist

I don't understand where this `identity_providers.ip_id` came from as I didn't type it anywhere.

I expect this item : `{user_id:1, ip_id:6, provider_id:'TESTID'}` to be inserted in users_identity_providers

Can you help ?

**Comment by koskimas** [comment]
https://vincit.github.io/objection.js/recipes/composite-keys.html#composite-keys

**Comment by Thomas-git** [comment]
I've already read the doc.
It doesn't help me to reference properly the composite key in the join through nor in the upsert. Can you provide an exemple ?

**Comment by Thomas-git** [comment]
Ok I've solded one part of the problem. 1) a typo in the update query. 2) I missed the "relate:true" property for graphOptions.
Now, another error appeared : 
On my IdentityProviders model, if I add "additionalProperties": false in the jsonSchema, I got error:
`ValidationError: providers[0].provider_id: is an invalid additional property, providers[0].id: is an invalid additional property`

But if I allow additionalProperties in jsonSchema, all works perfectly ! (postgres backend)
It's not a big deal because postgres will catch additional properties. But that's a useless round-trip to DB server. Why is Objection doing that before inserting the "extra" properties of the through join in the pivot table ?
Do I miss another query param ?
